publicapi/interaction.go

```
@@ -677,6 +762,15 @@ func (api InteractionAPI) AdmireToken(ctx context.Context, tokenID persist.DBID)
 		return "", err
 	}
 
+	admire, err := api.GetAdmireByActorIDAndTokenID(ctx, userID, tokenID)
```

I think it'd be awesome if the admire token operation was idempotent, such that if an admire already exists, we just return the existing admire. That way, we save a call to the database to check if an admire exists which means we don't need the dataloader and the query to fetch a specific admire anymore. It also means we don't need to return an error when an admire already exists.

We can update the existing query to something like this (haven't actually tested if this is valid SQL, but should be similar):
```
insert into admires (id, feed_event_id, post_id, token_id, actor_id) values ($1, sqlc.narg('feed_event'), sqlc.narg('post'), sqlc.narg('token'), $2)
on conflict (token_id, actor_id) do nothing
returning id;
```

The change is the `on conflict...` for the specific constraint (token_id, actor_id) that is being violated. I'd double-check if `returning id` returns the existing id, but I'm pretty sure it does!

It'd actually be awesome to do this for all the other things we can admire (post_id, feed_event_id), but feel free to skip unless its easy to also do!